### PR TITLE
Add `otherParameters` to ContentTypeRange, update `MediaType#matches` to take into account `otherParameters`.

### LINKED ISSUE #305: `MediaType#matches` ignores parameters.
Originally posted in https://github.com/softwaremill/tapir/issues/3232.

### PR DESCRIPTION
**Pull request by kamilkloch**
Fixes #305.

### CODE REVIEW

File: core/src/main/scala/sttp/model/ContentTypeRange.scala [hunk 1]
```diff
@@ -2,13 +2,24 @@ package sttp.model
 
 import sttp.model.ContentTypeRange.Wildcard
 
-case class ContentTypeRange(mainType: String, subType: String, charset: String) {
+case class ContentTypeRange(mainType: String, subType: String, charset: String, otherParameters: Map[String, String]) {
```
Comment (adamw) [inline core/src/main/scala/sttp/model/ContentTypeRange.scala:5]: this will for sure need some massaging to meet bin compat

Comment (kamilkloch) [reply core/src/main/scala/sttp/model/ContentTypeRange.scala:5]: Add constructor for the former case?

Comment (adamw) [reply core/src/main/scala/sttp/model/ContentTypeRange.scala:5]: Yes, that might work. Running mima locally should determine if there's anything more to do.

Comment (kamilkloch) [reply core/src/main/scala/sttp/model/ContentTypeRange.scala:5]: It does not :/

Comment (adamw) [reply core/src/main/scala/sttp/model/ContentTypeRange.scala:5]: `mimaReportBinaryIssues` doesn't work for you? Ah, maybe you need to fetch tags as described here: https://github.com/softwaremill/tapir#contributing

Comment (kamilkloch) [reply core/src/main/scala/sttp/model/ContentTypeRange.scala:5]: Do you have any pattern on how to resolve the issue with `copy`?

Comment (adamw) [reply core/src/main/scala/sttp/model/ContentTypeRange.scala:5]: Yes, you need to spell out the copy methods by hand, e.g.: https://github.com/softwaremill/tapir/blob/7cd79a1cf69cd2465841062abeb4b5ea988426c1/core/src/main/scala/sttp/tapir/EndpointIO.scala#L584

**Comment by kamilkloch** [comment]
I checked tapir locally against the locally published sttp-model. What do you mean by submitting this as a PR? It will not compile (missing locally published sttp-model).

**Comment by adamw** [comment]
I know, but some more changes are needed on the tapir side, no? :) I can then checkout both PRs locally & test

**Comment by kamilkloch** [comment]
I did not need to alter a single line in tapir to make it compile, I beileve it should work as expected with the new sttp-model. I tried to add some tests, but got lost in the tapir testing machinery :/ (`ContentNegitiation`, `ServerContentNegotiationTests`). I would appreciate your help here. Thank you!

**Comment by adamw** [comment]
`ServerContentNegotiationTests` is the right place to add a test :) If you add a test there, it will be ran when the tests for any specific interpreter's tests are run, e.g. `NettyFutureServerTest`.

The test itself consists of an endpoint (here you would need to create an endpoitn with the `oneOf`) and an sttp-client request to make (here you would need to include the appropriate header).

**Comment by kamilkloch** [comment]
@adamw please take a look: https://github.com/softwaremill/tapir/pull/3243

**Comment by adamw** [comment]
1.7.3 should be out soon

**Comment by adamw** [comment]
Yeah maybe simply update the PR that is already there with the test?

**Pull request merged**